src/instructlab/sdg/pipeline.py

```diff
@@ -131,15 +131,21 @@ def from_file(cls, ctx, pipeline_yaml):
             pipeline_yaml = os.path.join(resources.files(__package__), pipeline_yaml)
         return cls(ctx, pipeline_yaml, *_parse_pipeline_config_file(pipeline_yaml))
 
-    def generate(self, dataset) -> Dataset:
+    def generate(self, dataset, checkpoint_name=None) -> Dataset:
         """
         Generate the dataset by running the pipeline steps.
         dataset: the input dataset
+        checkpoint_name: unique subdir name for the checkpoint within checkpoint_dir
```

Comment:
I might have called this `run_id` or something - a unique name for this pipeline run - the only current use being for checkpoint directories